data/src/main/java/org/apache/iceberg/data/DeleteFilter.java

```diff
@@ -189,7 +192,11 @@ protected boolean shouldKeep(T item) {
     if (posDeletes.stream().mapToLong(DeleteFile::recordCount).sum() < setFilterThreshold) {
       return Deletes.filter(
           records, this::pos,
-          Deletes.toPositionSet(dataFile.path(), CloseableIterable.concat(deletes)));
+          Deletes.toPositionSet(
+              dataFile.path(),
+              CloseableIterable.concat(deletes, ThreadPools.getWorkerPool(), ThreadPools.getPoolParallelism())
```

Comment:
I think that parallelizing this requires adding an option. There are many cases where we wouldn't want to parallelize and cases where we want to use a thread pool besides the shared worker pool. It would be better to pass a pool in and parallelize if it is non-null.

Reply:
+1, something like `read.delete.position.num-threads` table property can work.

In general we follow this pattern when configuring and creating thread pools and executor service: https://github.com/apache/iceberg/blob/master/spark/src/main/java/org/apache/iceberg/spark/actions/BaseRewriteDataFilesSparkAction.java#L187-L194

Reply:
LGTM, I will trying to make it configurable.

Reply:
I add an optional method to set the read pool and config the read parallelism by code.  @rdblue 
And I think we could config the read parallelism by 'iceberg.worker.num-threads' rather than table property, because I don't want to spread the table property widely.  @jackye1995